### ISSUE TITLE
Add support for temporary credentials

### ISSUE THREAD

**Issue by et304383** [aws-amplify/amplify-cli#1420]
**Describe the solution you'd like**
Amplify CLI should support temporary credentials set as environment variables in the terminal.  Every other tool and SDK under the sun supports these.

**Describe alternatives you've considered**
Using profiles defined in ~/.aws/credentials with the temporary credentials that I have to update every 12 hours.

**Additional context**
It seems Amplify CLI only supports a profile OR pasting your IAM credentials from an IAM user into the configure prompt.

This is especially cumbersome when one is used to a workflow that uses a tool like aws-runas.

**Comment by UnleashedMind** [comment]
If you configure the Amplify CLI to use 'general' config level on your project, then the CLI will not configure credentials for the aws sdk it internally uses. This allows the aws sdk to resolve credentials by itself, and environmental variables will be used. [ref](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/setting-credentials-node.html)

This can be done in two ways: 
1. headless mode: 
Run `amplify init` or `amplify configure project` in headless mode, and set the `configLevel`  in `AWSCLOUDFORMATIONCONFIG` as `general`. check this [example](https://aws-amplify.github.io/docs/cli/init#sample-script).

2. Manually execute`amplify configure project` on a project already init'ed by the Amplify CLI,
and when prompted `Do you want to update or remove the project level configuration`
select `remove`

**Comment by et304383** [comment]
Thanks for the tips.  Is this documented somewhere?

**Comment by UnleashedMind** [comment]
It is explanation in this [doc](https://aws-amplify.github.io/docs/cli/init#--providers).

**Comment by cwaldbieser** [comment]
The links in this issue appear to be a bit out of date.  I think I found the relevant docs at https://docs.amplify.aws/cli/usage/headless#sample-script .

That said, I thought the whole point of amplify was to simplify the building process. Where I work, we have fully embraced Control Tower and the whole AWS multi-account strategy.  We don't allow our solutions teams to create IAM users except in very specific circumstances-- mainly we use the temporary credentials we get from the account portal (ala AWS SSO) if we need to run something via the CLI.

The fact that I have to cobble together a shell script to say I want to use the access key ID, secret access key, and session token is really kind of unwieldy.  I tried running this from the CLI, and `amplify` was able to create some resources in my target account, but then the script failed with "Failed to get profile: Profile configuration is missing for: undefined".  As a newcomer, this doesn't fill me with a lot of confidence that the process is set up correctly.  Trying to run `amplify push` just kind of sits there and spins after that.  Doesn't look like there is an actual network connection if I look for one in another terminal window.

Am I missing something?  Is there some more straightforward way I am supposed to run this using temporary credentials?
